code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -177,3 +177,92 @@
 	minbodytemp = 0
 	melee_damage_lower = 5
 	melee_damage_upper = 15
+
+/mob/living/simple_animal/pet/cat/cak
+	name = "Keeki"
+	desc = "It's a cat made out of cake. It looks like a brain would fit into it's head."
+	icon_state = "cak"
+	icon_living = "cak"
+	icon_resting = "cak_rest"
+	icon_dead = "cak_dead"
+	health = 50
+	maxHealth = 50
+	harm_intent_damage = 10
+	butcher_results = list(
+		/obj/item/organ/internal/heart = 1, 
+		/obj/item/reagent_containers/food/snacks/birthdaycakeslice = 3,  
+		/obj/item/reagent_containers/food/snacks/meat = 2
+	)
+	response_harm = "takes a bite out of"
+	attacked_sound = "sound/items/eatfood.ogg"
+	deathmessage = "loses its false life and collapses!"
+	death_sound = "bodyfall"
+	var/obj/item/organ/internal/brain/brain = null

```

Comment:
I really think utilizing the built in functionality of `CheckParts` would be better than this, to be honest.

```
/mob/living/simple_animal/pet/cat/cak/CheckParts(list/parts)
	..()
	var/obj/item/organ/brain/B = locate(/obj/item/organ/brain) in contents
	if(!B || !B.brainmob || !B.brainmob.mind)
		return
	B.brainmob.mind.transfer_to(src)
	to_chat(src, "<span class='big bold'>You are a cak!</span><b> You're a harmless cat/cake hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
	free cake to the station!</b>")
	var/new_name = stripped_input(src, "Enter your name, or press \"Cancel\" to stick with Keeki.", "Name Change")
	if(new_name)
		to_chat(src, "<span class='notice'>Your name is now <b>\"new_name\"</b>!</span>")
		name = new_name
```